src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactTestRendererFiber
+ * @preventMunge
+ * @flow
+ */
+
+'use strict';
+
+var ReactFiberReconciler = require('ReactFiberReconciler');
+var ReactGenericBatching = require('ReactGenericBatching');
+var emptyObject = require('emptyObject');
+
+import type { TestRendererOptions } from 'ReactTestMount';
+
+type ReactTestRendererJSON = {
+  type: string,
+  props: { [propName: string]: string },
+  children: null | Array<string | number | ReactTestRendererJSON>,
+  $$typeof?: any
+}
+
+let instanceCounter = 0;
+
+class TestContainer {
+  createNodeMock: Function;
+
+  constructor(createNodeMock) {
+    this.createNodeMock = createNodeMock;
+  }
+  appendChild(child) {}
+  insertBefore(beforeChild, child) {}
+  removeChild(child) {}
+  toJSON() {}
```

Comment:
Hmm...this actually seems to *maybe* work okay already? Here is my current test case:

```jsx
    it('can update text nodes when rendered as root', () => {
      var Component = (props) => props.children;

      var renderer = ReactTestRenderer.create(<Component>Hi</Component>);
      expect(renderer.toJSON()).toEqual('Hi');
      renderer.update(<Component>{['Hi', 'Bye']}</Component>);
      expect(renderer.toJSON()).toEqual('Hi Bye');
      renderer.update(<Component>Bye</Component>);
      expect(renderer.toJSON()).toEqual('Bye');
      renderer.update(<Component>{42}</Component>);
      expect(renderer.toJSON()).toEqual(42);
      renderer.update(<Component><div /></Component>);
      expect(renderer.toJSON()).toEqual({
        type: 'div',
        children: null,
        props: {},
      });
    });
```

Though currently `<Component>{['Hi', 'Bye']}</Component>` `toJSON()` only returns `Hi` (instead of `Hi Bye`). I’m not actually certain what is expected at this point.

I’m going to add another test case later, family-time!